File: insight-brain-data/src/main/java/com/sonatype/insight/brain/model/configuration/SystemNotice.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.insight.brain.model.configuration;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.sonatype.insight.model.HasStringId;

/**
 * @since 1.29.0
 */
@Entity
@Table(name = "system_notice")
public class SystemNotice
    implements HasStringId
{
  @Id
  @Column(name = "system_notice_id")
  private String id;

  @Column(name = "message")
  private String message = "";

  @Column(name = "enabled")
  private boolean enabled;

  @Override
  public String getId() {
    return id;
  }

  @Override
  public void setId(final String id) {
    this.id = id;
  }

  public String getMessage() {
    return message;
  }

  public void setMessage(final String message) {
    this.message = message;
  }

  public boolean isEnabled() {
    return enabled;
  }

  public void setEnabled(final boolean enabled) {
    this.enabled = enabled;
  }
}

File: insight-brain-service/src/main/frontend/systemNotice/systemNotice.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
function systemNoticeController(systemNoticeService, $scope) {
  var vm = this;

  vm.$onInit = function() {
    systemNoticeService.getSystemNotice().then(function(response) {
      vm.systemNotice = response;
    }).catch(function() {
      vm.systemNotice = systemNoticeService.getDefaultSystemNotice();
    });
  };

  $scope.$on('systemNoticeUpdated', function(systemNoticeUpdated, systemNotice) {
    vm.systemNotice = systemNotice;
  });
}

var systemNotice = {
  templateUrl: 'systemNotice/systemNotice.html?' + clmBuildTimestamp,
  controller: systemNoticeController,
  controllerAs: 'vm'
};

systemNoticeController.$inject = ['systemNoticeService', '$scope'];

angular.module('systemNoticeModule').component('systemNotice', systemNotice);
